lib/actionizer.rb

```diff
@@ -58,8 +58,11 @@ def method_missing(method_name, *args, &block)
       raise ArgumentError, "#{action_class.name}##{underlying_method}'s result must respond to :failure?"
     end
 
-    error = result.respond_to?(:error) ? result.error : "Unknown: Your result doesn't respond to :error"
-    fail!(error: error) if result.failure?
+    errors = result.to_h.select { |r| r.to_s.start_with? 'error' }
```

Comment:
Can you change the block variables to denote it's a hash? `{ |k, _v| k.to_s.start_with? 'error' }`